webapp/channels/src/components/announcement_bar/text_dismissable_bar.tsx

```diff
@@ -20,18 +20,20 @@ interface Props extends Partial<AnnouncementBarProps> {
     className?: string;
 }
 
-const TextDismissableBar = (props: Props) => {
-    const {allowDismissal, text, ...extraProps} = props;
-    const [dismissed, setDismissed] = useState<boolean>(false);
+const options = {
+    singleline: true,
+    mentionHighlight: false,
+};
+
+const TextDismissableBar = ({allowDismissal, text, onDismissal, ...extraProps}: Props) => {
+    const getDismissed = (text?: React.ReactNode) => localStorage.getItem(localStoragePrefix + text?.toString()) === 'true';
```

Comment:
This doesn't depend on anything inside the component, so it is safe to define it outside of the component (that is the reason we pass the text argument).

Reply:
Completed